src/utils/TextToSpeech.js

```diff
@@ -0,0 +1,74 @@
+const supportedVoiceLanguages = ['en', 'it'];
+let recommendedVoices = [];
+supportedVoiceLanguages.forEach(supportedVoiceLanguage => {
+  recommendedVoices[
+    supportedVoiceLanguage
+  ] = require(`./textToSpeech/${supportedVoiceLanguage}.json`);
+});
+
+export function hasTextToSpeechVoices() {
+  return speechSynthesis.getVoices().length !== 0;
+}
+
+export function startTextToSpeech(text, locale, onComplete) {
+  const voices = speechSynthesis.getVoices();
+  if (voices.length === 0) {
+    return false;
+  }
+
+  const filteredVoices = filterAvailableVoices(recommendedVoices[locale]);
+  if (filteredVoices.length === 0) {
+    return false;
+  }
+
+  const utterance = new SpeechSynthesisUtterance(text);
+
+  for (const voice of voices) {
+    if (voice.name === filteredVoices[0].name) {
+      utterance.voice = voice;
+      utterance.lang = voice.lang;
+      break;
+    }
+  }
+
+  utterance.onend = onComplete;
+
+  speechSynthesis.cancel();
+
+  speechSynthesis.speak(utterance);
+
+  console.log('Saying:', text);
+
+  return true;
+}
+
+export function stopTextToSpeech() {
+  speechSynthesis.cancel();
+}
+
+// From https://github.com/HadrienGardeur/web-speech-recommended-voices.
+function filterAvailableVoices(jsonData) {
+  if (!jsonData) {
+    return [];
+  }
+
+  const availableVoices = [];
+  const voices = speechSynthesis.getVoices();
+
+  jsonData.voices.forEach(function(voice) {
+    if (voices.some(apiVoice => apiVoice.name === voice.name)) {
+      availableVoices.push(voice);
+    } else {
+      if (voice.altNames) {
+        voice.altNames.forEach(function(altName) {
+          if (voices.some(apiVoice => apiVoice.name === altName)) {
+            voice.name = altName;
+            availableVoices.push(voice);
+          }
+        });
+      }
+    }
+  });
+
+  return availableVoices;
```

Comment:
I was thinking there might be some value in returning their array of results, in case we want to offer something like a voice selection UI, or allow the calling code to use additional selection heuristics.

Agreed on the confusion.  And also concerned by the way `voice` gets modified in the `altNames` case!  This code was borrowed direct from their demo, partly with the thought that improvements made there could be easily ported over.

Reply:
Ah got it, good to know it's copied over. Good thought re: there maybe being value in returning an array of results, hadn't thought of that!

Reply:
Looking at these a bit more, distinguishing `browserVoices` vs. `voicesMetadata` or something along those lines in var names might be helpful for future readers.